libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx

```diff
@@ -554,8 +554,18 @@ export class GlobalExplanationTab extends React.PureComponent<
     if (typeof item?.key === "string") {
       const key = item.key;
       const index = this.context.jointDataset.metaDict[key].index;
-      if (index !== undefined) {
-        this.handleFeatureSelection(this.state.selectedCohortIndex, index);
+      let indexAfterDrop;
+      for (const i of this.featureIndexMap.keys()) {
+        if (index && this.featureIndexMap.get(i) === index) {
+          indexAfterDrop = i;
+          break;
+        }
+      }
+      if (indexAfterDrop !== undefined) {
```

Comment:
nit: why not just call this inside the if and then break?  then you won't need to keep track of indexAfterDrop at all.  Maybe I am missing something here.

Reply:
Great comment. Updated!

Reply:
Thanks!